### PR TITLE
Don't display thumbnails for restricted works

### PR DESCRIPTION
**Pull request by warrd**

### CODE REVIEW

File: pipeline/transformer/transformer_mets/src/main/scala/uk/ac/wellcome/platform/transformer/mets/transformer/MetsData.scala [hunk 1]
```diff
@@ -111,16 +111,28 @@ case class MetsData(
...
+        .find(ImageUtils.isThumbnail)
+        .flatMap(ref => ImageUtils.buildThumbnailUrl(bnumber, ref))
+        .map { url =>
+          DigitalLocation(
+            url = url,
+            locationType = LocationType("thumbnail-image"),
+            license = license
+          )
+        } else
+      None
```
Comment (jamieparkinson) [inline pipeline/transformer/transformer_mets/src/main/scala/uk/ac/wellcome/platform/transformer/mets/transformer/MetsData.scala]: You can write this quite nicely as:

```suggestion
    for {
      fileReference <- fileReferences.find(ImageUtils.isThumbnail)
      url <- ImageUtils.buildThumbnailUrl(bnumber, fileReference)
      if accessStatus.forall(shouldCreateThumbnail)
    } yield
      DigitalLocation(
        url = url,
        locationType = LocationType("thumbnail-image"),
        license = license
      )
```

Comment (jamieparkinson) [reply pipeline/transformer/transformer_mets/src/main/scala/uk/ac/wellcome/platform/transformer/mets/transformer/MetsData.scala]: New GitHub feature 💯

Comment (warrd) [reply pipeline/transformer/transformer_mets/src/main/scala/uk/ac/wellcome/platform/transformer/mets/transformer/MetsData.scala]: Yeh I prefer the other method, I just find it easier to read for some reason :laughing:. Changed to the `for` syntax as not too bothered tho

Comment (warrd) [reply pipeline/transformer/transformer_mets/src/main/scala/uk/ac/wellcome/platform/transformer/mets/transformer/MetsData.scala]: Oh what didn't notice the `if` inside the `for` expression, thats freaky :alien:

**Pull request merged**